src/libs/actions/Policy.ts

```diff
@@ -1529,20 +1529,26 @@ function buildOptimisticPolicyRecentlyUsedCategories(policyID?: string, category
     return lodashUnion([category], policyRecentlyUsedCategories);
 }
 
-function buildOptimisticPolicyRecentlyUsedTags(policyID?: string, tag?: string): RecentlyUsedTags {
-    if (!policyID || !tag) {
+function buildOptimisticPolicyRecentlyUsedTags(policyID?: string, reportTags?: string): RecentlyUsedTags {
+    if (!policyID || !reportTags) {
         return {};
     }
 
     const policyTags = allPolicyTags?.[`${ONYXKEYS.COLLECTION.POLICY_TAGS}${policyID}`] ?? {};
-    // For now it only uses the first tag of the policy, since multi-tags are not yet supported
-    const tagListKey = Object.keys(policyTags)[0];
+    const policyTagKeys = Object.keys(policyTags);
     const policyRecentlyUsedTags = allRecentlyUsedTags?.[`${ONYXKEYS.COLLECTION.POLICY_RECENTLY_USED_TAGS}${policyID}`] ?? {};
+    const newOptimisticPolicyRecentlyUsedTags: RecentlyUsedTags = {};
 
-    return {
-        ...policyRecentlyUsedTags,
-        [tagListKey]: lodashUnion([tag], policyRecentlyUsedTags?.[tagListKey] ?? []),
-    };
+    reportTags.split(CONST.COLON).forEach((tag, index) => {
+        if (!tag) {
+            return;
+        }
+
+        const tagListKey = policyTagKeys[index];
+        newOptimisticPolicyRecentlyUsedTags[tagListKey] = lodashUnion([tag], policyRecentlyUsedTags[tagListKey] ?? []);
```

Comment:
Can't you just do `[...new Set([...tag,  ...(policyRecentlyUsedTags[tagListKey] ?? [])])`?

Reply:
Done - 11b6422.